File: pandas/over_120/R/over120.R
# df_orders and df_products are the provided dataframes.
getOrdersOver120 <- function (df_orders, df_products) {
    df <- merge(df_orders, df_products, by='product_id')
    df['total_amount'] <- df$quantity * df$price
    ret <- subset(df, total_amount > 120)
    ret[order(ret$order_id),]
}

File: pandas/over_120/python/over_120.py
import pandas as pd

def get_orders_over_120(df_orders: pd.DataFrame, df_products: pd.DataFrame):
    df = pd.merge(left=df_orders, right=df_products, how='left', on='product_id')
    df['total_amount'] = df.quantity * df.price
    return df[['order_id', 'product_id', 'quantity', 'total_amount']][df.total_amount > 120]
